darkmoon/api/v1/metadata/views.py

```diff
@@ -367,22 +414,27 @@ async def hash_comparison(
         }
 
         results = await collection.find(search_query).to_list(length=length)
-        li = [MetadataEntity.parse_obj(item) for item in results]
+        li: list[MetadataEntity] = []
+        for item in results:
+            if "header_info" in item:
+                li.append(ExeMetadataEntity.parse_obj(item))
+            else:
+                li.append(DocMetadataEntity.parse_obj(item))
+
         if len(li) == 0:
-            obj = MetadataEntity(
+            obj = DocMetadataEntity(
                 _id=PydanticObjectId(),
                 name=[inputFileName],
                 file_type=[inputFileType],
-                operating_system=[],
-                source_iso_name=[],
+                operating_system=[""],
```

Comment:
An empty string in a list is never valid here. Why can't we return either a `None` or an empty list?

Reply:
requested this fix to another branch handling this endpoint